src/Track.h

```diff
@@ -237,10 +237,30 @@ class AUDACITY_DLL_API Track /* not final */
    , public AttachedTrackObjects
    , public std::enable_shared_from_this<Track> // see SharedPointer()
 {
+public:
+
+   //! For two tracks describes the type of the linkage
+   enum class LinkType : int {
+       None = 0, //< No linkage
+       Group, //< Tracks are grouped together
+       Aligned, //< Tracks are grouped and changes should be synchronized
+   };
```

Comment:
This enumeration persists with the attribute name "linked" both before and after your changes.  So consider the problems of downgrade (3.1 file opened in 3.0.x) and the opposite problem of upgrade.

Downgrade is all right because any non-zero value in the file will become boolean "true" so that preserves whether the track is a left channel of a pair, or not.  Upgrade is more complicated.

As it is written now, stereo tracks saved in 3.0.x always load as "Group" meaning in the (usual?) case of clips all aligned between tracks, drag bars will appear on the right, because boolean true was saved and is now interpreted as 1.

You could swap Group and Aligned, but that only makes the opposite problem in the (less usual?) case that a file was saved with non-aligned clips, which should show right channel drag bars but will not.

The complete solution?  Make 1 a reserved "legacy" value.  Enlarge Track::LinkConsistencyCheck() so it detects numerically whether clips correspond between channels, and change "legacy" to Group or Aligned.

And of course LinkConsistencyCheck must become a virtual function so Track does not depend on WaveTrack, and the override for WaveTrack must also call-through to the base class method.

Reply:
It's quite strange solution i would say, why do you expect that file loaded from old project and having corresponding clip borders should be without affordances on the right channel (remember that with affordances both cases are valid)? Why not we remove them when clips are aligned at runtime then, or check this condition after merging tracks? Seems like it's another question for UX experts.
I don't like idea of LinkConsistencyCheck changing object state at all, now it's just make state somewhat cosistent, i can understand that, but i don't see any reason to prefer 'Aligned' over 'Grouped'

Reply:
If a file is loaded from an old project, and all clip borders do correspond, as would be usual -- do you think it is better there are affordances on the right?  Wouldn't the user instead want them removed?  But then, how could they be removed, once they are present?

"Why don't we remove them when clips are aligned at runtime then, or check this condition after merging tracks?"  Maybe that won't be done automatically, but maybe the choice to do it if possible, should also be given to the user.

For instance, what happens if you record a stereo track, split it, do nothing else (or do edits that don't affect clip boundaries), then join channels again?  Affordances enter the right track, and can never go away again.

I think that is a gap in the design.  I think the problem of writing the function that will detect that affordances can be removed is a problem that will need to be solved.

Reply:
How do you like this idea instead:  the enumeration does have four values, not three.  The new values are 2 and 3.  Old files may have the value 1.  LinkConsistencyCheck will correct 1 to 2 or 3.

It can simply correct 1 always to Group (which is now 2).

But if after testing we decide that a more complicated procedure is needed to correct the value -- then we will do that, and we have already prepared by reserving 1 for the old, un-corrected value, distinct from the two new values.

Reply:
>how could they be removed, once they are present?

They can't be removed
 
![image](https://user-images.githubusercontent.com/20165230/129017618-4f60dbdd-5e75-4156-bcb6-3fcdf8aeb478.png)

>do you think it is better there are affordances on the right
I think that it's safest assumption (keep things as they were before)

>Maybe that won't be done automatically, but maybe the choice to do it if possible, should also be given to the user.

Well maybe, but that was not declared in the scope of work for this task

Reply:
I have a suspicion that the scope may enlarge after all to include this after some more testing, but I admit this is trying to read designers' minds right now.